enclave_wrangler/main.py

```diff
@@ -242,27 +270,75 @@ def run(input_csv_folder_path):
         # and if needed we can also save the json data in upd_cs_ver_expression_items_dict
         # premade_codeset_id is stored in the codeset_id column in the csv files, save the id in the enclave_codeset_id column
         # update when it was uploaded as well, Stephane 3/15/22
-        code_sets_df.set_index('codeset_id', inplace=True)
+        try:
+            code_sets_df.set_index('codeset_id', inplace=True)
+        except Exception as e:
+            print(e)
+
         code_sets_df.at[premade_codeset_id, 'enclave_codeset_id'] = codeset_id
         code_sets_df.at[premade_codeset_id, 'enclave_codeset_id_updated_at'] = _datetime_palantir_format()
-        code_sets_df.reset_index()
+        code_sets_df = code_sets_df.reset_index()
         # return response_json
 
 
     # write out the update csv file with the enclave_codeset_id
     # print('before terminating write out the updated code_sets.csv file here')
     # date_str = datetime.now().strftime('%Y_%m_%d_%H_%M')
     # output_filename = 'code_sets_updated_' + date_str + '.csv'
+    # TODO: fix creates these columns depending on how we're doing indexing: Unnamed: 0	Unnamed: 0.1, etc
     output_filename = 'code_sets.csv'
     code_sets_df.to_csv(os.path.join(input_csv_folder_path, output_filename), index=True, encoding='utf-8')
 
-    # Save to persistence layer
+    return code_sets_df
+
+
+def left_join_update(df, df2):
```

Comment:
@Sigfried @stephanieshong This is how I decided to solve the `LEFT JOIN` and corresponding field updates.

I guess it's not entirely inelegant, but I can't help but feel this should be a pretty standard use case and shouldn't require this much code to do, but maybe my intuition is just wrong, or perhaps there's a nicer possible implementation.